validators.go

```diff
@@ -0,0 +1,139 @@
+package imgix
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// validateDomain uses Go's url.Parse and url.Hostname functions to
+// validate the domain. Elsewhere we use a regex to filter invalid
+// domains. However, the same regex won't work in this case as Go
+// does not support positive look-a-heads (i.e. `(?=)`).
+// TODO: Discuss. Go doesn't support positive look-a-heads so our
+// domain regex won't work here. I will explore regex alternatives.
```

Comment:
So I think the best way to ensure this mimics domain validation elsewhere is to thoroughly test it. I know that the FE team has a series of tests for source creation that we can probably leverage here.

Reply:
Agreed!